waddrmgr/address.go

```diff
@@ -629,6 +629,52 @@ func newManagedAddress(s *ScopedKeyManager, derivationPath DerivationPath,
 	managedAddr.privKeyEncrypted = privKeyEncrypted
 	managedAddr.privKeyCT = privKeyBytes
 
+	// At this point, we've derived an address based on a private key which
+	// was the output of a BIP 32 derivation. As a sanity check, we'll make
+	// sure that we can properly generate a valid signature/witness for the
+	// given address type.
+	var msg [32]byte
+	if _, err := rand.Read(msg[:]); err != nil {
+		return nil, fmt.Errorf("unable to read random "+
+			"challenge for addr validation: %w", err)
+	}
+
+	// We'll first validate things against the private key we got
+	// from the original derivation.
+	err = managedAddr.Validate(msg, privKey)
+	if err != nil {
+		return nil, fmt.Errorf("addr validation for addr=%v "+
```

Comment:
nit: `Address` implements `Sting()`, is `%s` better here?

Reply:
I think the end result would be the same? Since `%v` will try to call `String` if it exists.